lib/pbench/test/unit/server/query_apis/test_query_builder.py

```diff
@@ -10,114 +12,110 @@ class TestQueryBuilder:
+                    {"term": {"authorization.owner": id}},
                     {"term": {"authorization.access": "public"}},
-                    {"term": {"authorization.owner": "drb"}},
                 ]
             }
         }
 
-    def test_user(self, elasticbase, server_config, current_user_drb, user_ok):
+    def test_user(self, elasticbase, server_config, current_user_drb):
         """
         Test the query builder when only a user is specified. We expect the
         query builder to add the user term. Note that in the real server we
         can reach the builder only if there is an authenticated user which
         can access datasets owned by the specified user, so it simply builds
         the query.
         """
+        id = str(current_user_drb.id)
         query: JSON = elasticbase._get_user_query(
-            {"user": "drb"}, [{"term": {"icecream": "vanilla"}}]
+            {"user": id}, [{"term": {"icecream": "chocolate"}}]
         )
         assert query == {
             "bool": {
                 "filter": [
-                    {"term": {"icecream": "vanilla"}},
-                    {"term": {"authorization.owner": "drb"}},
+                    {"term": {"icecream": "chocolate"}},
+                    {"term": {"authorization.owner": id}},
+                ]
+            }
+        }
+
+    def test_user_admin(self, elasticbase, server_config, current_user_admin):
+        """
+        Test the query builder when only a user is specified with administrator
+        authentication. We expect the query builder to add the user term.
+
+        This is the same code path as `test_user`, above, and doesn't add any
+        coverage, but "feels" distinct.
+        """
+        # Invent a user ID that's different from the generated admin user's ID;
+        # the value doesn't matter for this test.
+        id = str(current_user_admin.id + 1)
+        query: JSON = elasticbase._get_user_query(
+            {"user": id}, [{"term": {"icecream": "keylime"}}]
+        )
+        assert query == {
+            "bool": {
+                "filter": [
+                    {"term": {"icecream": "keylime"}},
+                    {"term": {"authorization.owner": id}},
                 ]
             }
         }
 
-    def test_access_noauth_private(
-        self, elasticbase, server_config, current_user_none, user_ok
-    ):
+    def test_access_noauth_private(self, elasticbase, server_config, current_user_none):
         """
         Test the query builder when "private" access is specified by an
-        unauthenticated client. This is expected to throw an exception, which
-        will be treated as if Elasticsearch returned no hits.
+        unauthenticated client. This is an "impossible" case as authorization
+        checks will stop the request and return UNAUTHORIZED before we get
+        here. We cover the code path anyway, looking for the AssemblyError
+        exception.
```

Comment:
I don't think that this comment should be describing code elsewhere which it isn't testing.  That is, the code here doesn't care that the setup yields an "impossible" case -- we're gonna do the set up and check that it produces the result that we want anyway.  I think you should basically delete the middle sentence and the first half of the last sentence.

Reply:
This comment now applies to lines 140-142 in `test_auth_access_private()` and lines 165-167 in `test_noauth_access_private()`.